<?xml version="1.0"?>
<robot name="cart_pendulum">

  <link name="world"/>

  <joint name="world_static_joint" type="fixed">
      <origin rpy="0 0 0" xyz="0 0 2"/>
      <parent link="world"/>
      <child link="slide_bar"/>
  </joint>

  <link name="slide_bar">
    <visual>
      <origin xyz="0 0 0"/>
      <geometry>
        <box size="15 0.005 0.005"/>
      </geometry>
      <material name="grey">
        <color rgba="0.75 0.75 0.75 1.0"/>
      </material>
    </visual>
      <inertial>
        <mass value="3"/>
        <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyy="1.0" iyz="0.0" izz="1.0"/>
      </inertial>
  </link>

  <joint name="slider_to_cart" type="prismatic">
    <axis xyz="-1 0 0"/>
    <origin xyz="0.0 0.0 0.0"/>
    <parent link="slide_bar"/>
    <child link="cart"/>
    <limit effort="100000.0" lower="-150" upper="150" velocity="50"/>
    <dynamics damping="0.0" friction="100.0"/>
  </joint>

  <link name="cart">
    <visual>
      <origin xyz="0 0 0"/>
      <geometry>
        <box size="0.05 0.01 0.01"/>
      </geometry>
      <material name="blue">
        <color rgba="0.0 0 1.0 1.0"/>
      </material>
    </visual>
    <collision>
      <origin xyz="0 0 0"/>
      <geometry>
        <box size="0.1 0.05 0.05"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="3"/>
      <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyy="1.0" iyz="0.0" izz="1.0"/>
    </inertial>
  </link>

  <joint name="cart_to_stick" type="continuous">
    <axis xyz="0 1 0"/>
    <origin xyz="0.0 0.015 0.0" rpy="0 -1.57 0"/>
    <parent link="cart"/>
    <child link="stick"/>
    <!-- <limit effort="100000.0" lower="-150" upper="150" velocity="50"/> -->
    <dynamics damping="0.0" friction="100.0"/>
  </joint>

  <link name="stick">
    <visual>
      <origin xyz="0.125 0.0 0.0"/>
      <geometry>
        <box size="0.25 0.01 0.01"/>
      </geometry>
      <origin xyz="0 0 0"/>
      <!-- <material name="red"> -->
        <!-- <color rgba="1.0 0 0.0 1.0"/> -->
      <!-- </material> -->
      <material name="grey">
        <color rgba="0.75 0.75 0.75 1.0"/>
      </material>
    </visual>
    <collision>
      <origin xyz="0.125 0.0 0.0"/>
      <geometry>
        <box size="0.25 0.01 0.01"/>
      </geometry>
      <origin xyz="0 0 0"/>
    </collision>
    <inertial>
      <origin xyz="0.125 0.0 0.0"/>
      <mass value="0.0"/>
      <inertia ixx="0.0" ixy="0.0" ixz="0.0" iyy="0.0" iyz="0.0" izz="0.0"/>
    </inertial>
  </link>

    <joint name="stick_to_ball_joint" type="fixed">
      <origin rpy="0 0 0" xyz="0.25 0 0"/>
      <parent link="stick"/>
      <child link="ball"/>
  </joint>

  <link name="ball">
    <visual>
      <geometry>
        <sphere radius="0.02"/>
      </geometry>
      <material name="red">
        <color rgba="1.0 0 0 1.0"/>
      </material>
    </visual>
      <inertial>
        <mass value="1"/>
        <inertia ixx="4e-05" ixy="0.0" ixz="0.0" iyy="4e-05" iyz="0.0" izz="4e-05"/>
      </inertial>
  </link>

</robot>
